server/sonar-server/src/main/java/org/sonar/server/user/index/UserIndexer.java

```diff
@@ -58,52 +61,104 @@ public void indexOnStartup(Set<IndexType> emptyIndexTypes) {
     doIndex(newBulkIndexer(Size.LARGE), null);
   }
 
+  public void commitAndIndex(DbSession dbSession, UserDto user) {
+    EsQueueDto queueItem = dbClient.esQueueDao().insert(dbSession, EsQueueDto.create(EsQueueDto.Type.USER, user.getLogin()));
+    dbSession.commit();
+    postCommit(dbSession, user, queueItem);
+  }
+
+  /**
+   * Entry point for Byteman tests. See directory tests/resilience.
+   * The unused parameter "user" is used only by the Byteman script.
+   */
+  private void postCommit(DbSession dbSession, UserDto user, EsQueueDto queueItem) {
+    index(dbSession, singletonList(queueItem));
+  }
+
+  public void index(DbSession dbSession, Collection<EsQueueDto> esQueueItems) {
+    if (esQueueItems.isEmpty()) {
+      return;
+    }
+    esQueueItems.stream()
+      .forEach(i -> requireNonNull(i.getUuid(), () -> "BUG - " + i + " has not been persisted before indexing"));
+
+    BulkIndexer bulkIndexer = newBulkIndexer(Size.REGULAR);
+    try (UserResultSetIterator rowIt = UserResultSetIterator.create(dbClient, dbSession, esQueueItems)) {
+      processResultSet(dbSession, bulkIndexer, esQueueItems, rowIt);
+    }
+  }
+
+  @Deprecated
   public void index(String login) {
-    requireNonNull(login);
     doIndex(newBulkIndexer(Size.REGULAR), singletonList(login));
   }
 
+  @Deprecated
   public void index(List<String> logins) {
-    requireNonNull(logins);
-    if (logins.isEmpty()) {
-      return;
+    if (!logins.isEmpty()) {
+      doIndex(newBulkIndexer(Size.REGULAR), logins);
     }
+  }
 
-    doIndex(newBulkIndexer(Size.REGULAR), logins);
+  @Deprecated
+  public void index(DbSession dbSession, List<String> logins) {
+    if (!logins.isEmpty()) {
+      doIndex(dbSession, newBulkIndexer(Size.REGULAR), logins);
+    }
   }
 
+  @Deprecated
   private void doIndex(BulkIndexer bulk, @Nullable List<String> logins) {
     try (DbSession dbSession = dbClient.openSession(false)) {
-      if (logins == null) {
-        processLogins(bulk, dbSession, null);
-      } else {
-        executeLargeInputsWithoutOutput(logins, l -> processLogins(bulk, dbSession, l));
-      }
+      doIndex(dbSession, bulk, logins);
     }
   }
 
+  @Deprecated
+  private void doIndex(DbSession dbSession, BulkIndexer bulk, @Nullable List<String> logins) {
+    if (logins == null) {
+      processLogins(bulk, dbSession, null);
+    } else {
+      executeLargeInputsWithoutOutput(logins, l -> processLogins(bulk, dbSession, l));
+    }
+  }
+
+  @Deprecated
   private void processLogins(BulkIndexer bulk, DbSession dbSession, @Nullable List<String> logins) {
-    try (UserResultSetIterator rowIt = UserResultSetIterator.create(dbClient, dbSession, logins)) {
-      processResultSet(bulk, rowIt);
+    List<EsQueueDto> esQueueDtos = null;
+
+    if (logins != null) {
+      esQueueDtos = logins.stream()
+        .map(l -> {
+          EsQueueDto esQueueItem = EsQueueDto.create(EsQueueDto.Type.USER, l);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this line start at column 32. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AIndentationCheck)